### ISSUE TITLE
TypeError: Cannot read property 'hasOwnProperty' of null

### ISSUE THREAD

**Issue by aaraggornn** [almende/vis#2679]
Upgrading from 4.17 to 4.18.1 I get the following error:
```
error_handler.js:60 TypeError: Cannot read property 'hasOwnProperty' of null
    at Object.e.selectiveExtend (eval at webpackJsonp.271.module.exports (http://localhost:4200/scripts.bundle.js:25:8), <anonymous>:25:2701)
    at new n (eval at webpackJsonp.271.module.exports (http://localhost:4200/scripts.bundle.js:25:8), <anonymous>:37:23232)
    at o (eval at webpackJsonp.271.module.exports (http://localhost:4200/scripts.bundle.js:25:8), <anonymous>:37:18477)
    at n.eval (eval at webpackJsonp.271.module.exports (http://localhost:4200/scripts.bundle.js:25:8), <anonymous>:36:11274)
    at Array.forEach (native)
    at n._onUpdate (eval at webpackJsonp.271.module.exports (http://localhost:4200/scripts.bundle.js:25:8), <anonymous>:36:10856)
    at n.setItems (eval at webpackJsonp.271.module.exports (http://localhost:4200/scripts.bundle.js:25:8), <anonymous>:36:9343)
    at n.setItems (eval at webpackJsonp.271.module.exports (http://localhost:4200/scripts.bundle.js:25:8), <anonymous>:28:4099)
    at n (eval at webpackJsonp.271.module.exports (http://localhost:4200/scripts.bundle.js:25:8), <anonymous>:28:3126)
    at VisTimelineCategorizedDirective.webpackJsonp.918.VisTimelineCategorizedDirective.generateChart (http://localhost:4200/main.bundle.js:4845:25)
    at VisTimelineCategorizedDirective.webpackJsonp.918.VisTimelineCategorizedDirective.reload (http://localhost:4200/main.bundle.js:4251:18)
    at SafeSubscriber._next (http://localhost:4200/main.bundle.js:4341:19)
    at SafeSubscriber.__tryOrUnsub (http://localhost:4200/vendor.bundle.js:4897:16)
    at SafeSubscriber.next (http://localhost:4200/vendor.bundle.js:4846:22)
    at Subscriber._next (http://localhost:4200/vendor.bundle.js:4799:26)
```
Downgrading back to 4.17 works correctly. My project is an Angular2 project.
I create the Timeline with the following line: 
```
this.timeline = new vis.Timeline(this.el.nativeElement, this.dataSets.active.items, this.dataSets.active.groups, options);
```
this.dataSets.active.items is a `new vis.DataSet<vis.DataItem>()` with groups a DataSet of DataGroup

**Comment by yotamberk** [comment]
@aaraggornn Can you supply a jsbin? It's difficult to reproduce without a working example of the bug

**Comment by aaraggornn** [comment]
Here is a jsbin with a version of what I'm currently using.
http://jsbin.com/gemitiy/edit?html,js,output
If you change the version to 4.17.0 in the html it will work as intended

**Comment by yotamberk** [comment]
Thanks! The problem is here: https://github.com/almende/vis/blob/bc2bc581aaa16905f2ee15b26d3915648c34cc54/lib/timeline/component/item/Item.js#L45

it should be `this.data.editable` instead of `data.editable`

We will fix this ASAP

**Comment by yotamberk** [comment]
Thanks! there was a mistake in there was a mistake in the code. I submitted a PR to fix this

**Comment by mojoaxel** [comment]
fixed by @yotamberk with #2735

**Comment by aaraggornn** [comment]
Using version 4.19.1, the supplied jsbin still has the same issue

**Comment by yotamberk** [comment]
@aaraggornn how can we reproduce this so we can fix it?

**Comment by yotamberk** [comment]
fixed in #2679
